File: youlexuan-cart-interface/src/main/java/com/offcn/cart/service/CartService.java
package com.offcn.cart.service;

import com.offcn.entity.Cart;

import java.util.List;

public interface CartService {

    /**
     * 添加商品到购物车
     */
    public List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);


    /**
     * 从redis中查询购物车列表
     */
    public List<Cart> findCartListFromRedis(String username);

    /**
     * 添加购物车到redis缓存中
     */
    public void saveCartListToRedis(String username,List<Cart> cartList);

    /**
     * 合并购物车 登陆后将cookie中的数据存入redis，清空cookie
     */
    public List<Cart> mergeCartList(List<Cart> cartList1,List<Cart> cartList2);
}

File: youlexuan-cart-web/src/main/webapp/js/service/cartService.js
app.service('cartService',function ($http) {

    this.findCartList=function () {
        return $http.get('../cart/findCartList.do');
    }

    this.addGoodsToCartList=function (itemId,num) {
        return $http.get('../cart/addCartList.do?itemId='+itemId+"&num="+num);
    }
    //求合计
    this.sum=function(cartList){
        var totalValue={totalNum:0, totalMoney:0.00 };//合计实体
        for(var i=0;i<cartList.length;i++){
            var cart=cartList[i];
            for(var j=0;j<cart.orderItemList.length;j++){
                var orderItem=cart.orderItemList[j];//购物车明细
                totalValue.totalNum+=orderItem.num;
                totalValue.totalMoney+= orderItem.totalFee;
            }
        }
        return totalValue;
    }

    // 根据用户id获取地址列表
    this.findAddressList=function () {
        return $http.get('../address/findAddressListByUserId.do');
    }

    //增加
    this.add=function(entity){
        return  $http.post('../address/add.do',entity );
    }
    //修改
    this.update=function(entity){
        return  $http.post('../address/update.do',entity );
    }
    //查询实体
    this.findOne=function(id){
        return $http.get('../address/findOne.do?id='+id);
    }
    //删除
    this.dele=function(id){
        return $http.get('../address/delete.do?id='+id);
    }

    //提交订单
    this.submitOrder=function (order) {
        return $http.post('../order/add.do',order);
    }
})